nixos/modules/services/backup/syncoid.nix

```diff
@@ -267,14 +267,14 @@ in
             # syncoid may need zpool to get feature@extensible_dataset
             path = [ "/run/booted-system/sw/bin/" ];
             serviceConfig = {
-              ExecStartPre =
+              ExecStartPre = mkIf (cfg.user != "root") (
```

Comment:
**This comment is outside the scope of this PR. I may make another PR for that later**
I think we should drop `cfg.user`, change `commands` to `listOf` instead of `attrsOf` and switch to `DynamicUser=` (which provides a temporary `$USER`), this would enable to send the same dataset to multiple targets and still use `zfs unallow` without a race-condition.